Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift

```diff
@@ -457,11 +483,22 @@ extension PlaceCarouselViewController: LocationMonitorDelegate {
     func locationMonitor(_ locationMonitor: LocationMonitor, userDidExitCurrentLocation location: CLLocation) {
         locationMonitor.refreshLocation()
     }
+
+    func locationMonitor(_ locationMonitor: LocationMonitor, didFailInitialUpdateWithError error: Error) {
+        // Note: actually, at this point, if the location comes, `didUpdateLocation` will be called.
+        presentNoLocationAlert()
+    }
 }
 
 extension PlaceCarouselViewController: PlacesProviderDelegate {
     fileprivate func showErrorMessageIfNoPlaces() {
         if self.places.isEmpty {
+            let hasLocation = locationMonitor.getCurrentLocation() != nil
+            guard hasLocation else {
+                presentNoLocationAlert()
```

Comment:
Since this method is fired after a timeout fetching places, is it possible for this to be called when an alert is already being shown because of an error? If so, we might want to check to see if we are already showing one before showing another.